homeassistant/components/light/httplight.py

```diff
@@ -0,0 +1,175 @@
+"""
+Support for lights that use the HTTP protocol.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/light.httplight/
+"""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.const import CONF_HOST, CONF_NAME, CONF_TYPE
+from homeassistant.components.light import (
+    Light, ATTR_BRIGHTNESS,
+    ATTR_HS_COLOR, SUPPORT_BRIGHTNESS,
+    SUPPORT_COLOR, PLATFORM_SCHEMA)
+import homeassistant.helpers.config_validation as cv
+import homeassistant.util.color as color_util
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_TYPE_SIMPLE = 'simple'
+CONF_TYPE_DIMMABLE = 'dimmable'
+CONF_TYPE_COLOR = 'color'
+
+CONF_TYPES = [CONF_TYPE_SIMPLE, CONF_TYPE_DIMMABLE, CONF_TYPE_COLOR]
+
+SUPPORT_HTTPLIGHT_BRIGHTNESS = SUPPORT_BRIGHTNESS
+SUPPORT_HTTPLIGHT_COLOR = SUPPORT_BRIGHTNESS | SUPPORT_COLOR
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Required(CONF_NAME): cv.string,
+    vol.Required(CONF_TYPE): vol.In(CONF_TYPES)
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the HTTP Light platform."""
+    import requests
+
+    host = config.get(CONF_HOST)
+    light_type = config.get(CONF_TYPE)
+    name = config.get(CONF_NAME)
+
+    try:
+        if light_type == CONF_TYPE_SIMPLE:
+            get_state = requests.get(
+                url='http://{}/state'.format(host), timeout=10)
+            state = bool(get_state.text.partition('\r\n')[0] == 'on')
+        else:
+            get_state = requests.get(
+                url='http://{}/brightness'.format(host), timeout=10)
+            brightness = get_state.text.partition('\r\n')[0]
+            state = bool(int(int(brightness) * 2.55) > 0)
+    except requests.exceptions.MissingSchema:
+        _LOGGER.error("Missing or invalid host in configuration.")
```

Comment:
`host` can't be missing. It's a requirement.

How can `MissingSchema` be triggered when you add the schema to the URL?